modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java

```diff
@@ -807,6 +841,8 @@ public void add(FeatureEntry entry, SimpleFeatureCollection collection) throws I
         FeatureEntry e = new FeatureEntry();
         e.init(entry);
 
+        collection = forceXY(collection);
```

Comment:
The origional geoserver issue indicated this was a common problem for GeoPackage and CSV; if we see more instances of forceXY being needed to would be kind to add a utility method to DataUtilities.

Reply:
That's certainly a possibility.  There are 2 reprojecting feature collections - one in GT and one in GS.  The one is GS operates a bit differently - it first uses the underlying Geometry's userdata (expecting to see a CRS) instead of the one in the Schema.

Reply:
Understood :)

I think if we do this for real as part of DataUtilities we would need to update the CoordianteReferenceSystem stored in the Geometry user data (for output formats such as GML that can benefit).